test/system/src/test/java/io/pravega/test/system/BasicCBRTest.java

```diff
@@ -0,0 +1,181 @@
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(clientConfig)
+                .maxBackoffMillis(5000).build(), executor);
+        streamManager = StreamManager.create(clientConfig);
+
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM,
+                StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.fixed(1))
+                        .retentionPolicy(RetentionPolicy.bySizeBytes(MIN, MAX)).build()));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+    }
+
+    @Test
+    public void basicCBRTest() throws Exception {
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(SCOPE, clientConfig);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write a single event.
+        log.info("Writing event e1 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent("e1", "data of size 30").join();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        readerGroupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(READER_GROUP);
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(READER_GROUP + "-" + String.valueOf(1),
+                READER_GROUP, new JavaSerializer<>(), readerConfig);
+
+        // Read one event and update the retention stream-cut.
+        log.info("Reading event e1 from {}/{}", SCOPE, STREAM);
+        reader.readNextEvent(READ_TIMEOUT);
+        log.info("{} generating stream-cuts for {}/{}", READER_GROUP, SCOPE, STREAM);
+        CompletableFuture<Map<Stream, StreamCut>> futureCuts = readerGroup.generateStreamCuts(executor);
+        EventRead<String> emptyEvent = reader.readNextEvent(100);
```

Comment:
What is this used for?

Reply:
In order to trigger the silent checkpoint generated during `generateStreamCuts` to complete we read another event after the `ReaderGroupState` updates itself.

Reply:
Fixed that.